src/Worker/PushSubscription.php

```diff
@@ -48,32 +52,57 @@ public static function execute(int $sid, int $nid)
 			return;
 		}
 
-		if (!empty($notification['uri-id'])) {
-			$notify = DBA::selectFirst('notify', ['msg'], ['uri-id' => $notification['target-uri-id']]);
+		$application_token = DBA::selectFirst('application-token', [], ['application-id' => $subscription['application-id'], 'uid' => $subscription['uid']]);
+		if (empty($application_token)) {
+			Logger::info('Application token not found', ['application' => $subscription['application-id']]);
+			return;
+		}
+
+		$user = User::getById($notification['uid']);
+		if (empty($user)) {
+			Logger::info('User not found', ['application' => $subscription['uid']]);
+			return;
 		}
 
+		$l10n = DI::l10n()->withLang($user['language']);
+
+		$type = Notification::getType($notification);
+
 		if (!empty($notification['actor-id'])) {
 			$actor = Contact::getById($notification['actor-id']);
 		}
 
-		$push = [
-			'subscription' => Subscription::create([
-				'endpoint'  => $subscription['endpoint'],
-				'publicKey' => $subscription['pubkey'],
-				'authToken' => $subscription['secret'],
-			]),
-			// @todo Check if we are supposed to transmit a payload at all
-			'payload' => json_encode([
-				'title'     => 'Friendica',
-				'body'      => $notify['msg'] ?? '',
-				'icon'      => $actor['thumb'] ?? '',
-				'image'     => '',
-				'badge'     => DI::baseUrl()->get() . '/images/friendica-192.png',
-				'tag'       => $notification['parent-uri-id'] ?? '',
-				'timestamp' => DateTimeFormat::utc($notification['created'], DateTimeFormat::JSON),
-			]),
+		$body = '';
+
+		if (!empty($notification['target-uri-id'])) {
+			$post = Post::selectFirst([], ['uri-id' => $notification['target-uri-id'], 'uid' => [0, $notification['uid']]]);
+			if (!empty($post['body'])) {
+				$body = BBCode::toPlaintext($post['body'], false);
+				$body = Plaintext::shorten($body, 160, $notification['uid']);
+			}
+		}
+
+		$push = Subscription::create([
+			'contentEncoding' => 'aesgcm',
+			'endpoint'        => $subscription['endpoint'],
+			'keys'            => [
+				'p256dh' => $subscription['pubkey'],
+				'auth'   => $subscription['secret']
+			],
+		]);
+
+		$payload = [
+			'access_token'      => $application_token['access_token'],
+			'preferred_locale'  => $user['language'],
+			'notification_id'   => $nid,
+			'notification_type' => $type,
+			'icon'              => $actor['thumb'] ?? '',
+			'title'             => $l10n->t('Notification from Friendica'), // @todo Replace it with a speaking title
```

Comment:
A "speaking" title?

Reply:
something meaningful I guess

Reply:
I mean a title like "XYZ has replied to your comment". We cannot use the ones from the "notify" table here since it is not yet created when we arrive here. And also the "notify" is part of the whole legacy code of the notification system, so we shouldn't use it at all.